project/tools/version.py

```diff
@@ -0,0 +1,92 @@
+"""
+Very simple heuristic to generate the next version number
+based on the current changelog news fragments.
+
+This looks for the most recent version by parsing the
+CHANGELOG.md file and increments a specific part,
+depending on the fragment types present and their contents.
+
+Major bumps are caused by:
+  * files named `.removed.md`
+  * files named `.breaking.md`
+  * files containing `BREAKING:`
+
+Minor bumps are caused by:
+  * files named `.added.md`
+
+Otherwise, only the patch version is bumped.
+"""
+
+import re
+import sys
+from pathlib import Path
+
+PROJECT_ROOT = Path(__file__).parent.parent.resolve()
```

Comment:
Suggestion: just use the current working directory. This way the script can be run in any directory without the hardcoded assumption about the folder structure.